snapcraft/pack.py

```diff
@@ -90,6 +90,64 @@ def _get_filename(
     return None
 
 
+def _pack(command: List[Union[str, Path]]) -> str:
```

Comment:
As it is, this feels like a pretty generic command wrapper, with the logic for creating the `pack` command still in `pack_components` or `pack_snap`. What about making `_pack` a bit smarter like:

```suggestion
def _pack(directory: Path, output_dir: Path, *, compression: Optional[str] = None, filename: Optional[str] = None) -> str:
```

Reply:
Great idea!